models/helper.go

```diff
@@ -0,0 +1,32 @@
+package models
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/uc-cdis/cohort-middleware/utils"
+	"gorm.io/gorm"
+)
+
+func QueryFilterByConceptIdsAndCohortPairsHelper(query *gorm.DB, filterConceptIds []int64, filterCohortPairs []utils.CustomDichotomousVariableDef, omopSchemaName string, resultSchemaName string) *gorm.DB {
+	for i, filterConceptId := range filterConceptIds {
+		observationTableAlias := fmt.Sprintf("observation_filter_%d", i)
+		log.Printf("Adding extra INNER JOIN with alias %s", observationTableAlias)
+		query = query.Joins("INNER JOIN "+omopSchemaName+".observation as "+observationTableAlias+" ON "+observationTableAlias+".person_id = observation.person_id").
+			Where(observationTableAlias+".observation_concept_id = ?", filterConceptId).
+			Where("(" + observationTableAlias + ".value_as_string is not null or " + observationTableAlias + ".value_as_number is not null)") // TODO - improve performance by only filtering on type according to getConceptValueType()
+	}
+
+	for i, filterCohortPair := range filterCohortPairs {
```

Comment:
the original code had a comment above this loop. I think the comment is still useful. Maybe add it again (some variable names did change - so adjust that in the comment as well)?

Reply:
fixed